File: src/GWallet.Frontend.XF/Initialization.fs
#if XAMARIN
namespace GWallet.Frontend.XF
#else
namespace GWallet.Frontend.Maui
#endif

open System.Linq

#if XAMARIN
open Xamarin.Forms
#else
open Microsoft.Maui
open Microsoft.Maui.Controls
#endif

open GWallet.Backend

module Initialization =

    let internal GlobalState = GlobalState()

    let private GlobalInit () =
        Infrastructure.SetupExceptionHook ()

    let internal LandingPage(): NavigationPage =
        GlobalInit ()

        let accounts = Account.GetAllActiveAccounts()
        let landingPage:Page =
            if not (accounts.Any()) then
                (WelcomePage GlobalState) :> Page
            else
                (LoadingPage (GlobalState, true)) :> Page

        let navPage = NavigationPage landingPage
        NavigationPage.SetHasNavigationBar(landingPage, false)
        navPage

File: src/GWallet.Frontend.XF.CS/Initialization.cs
using System;

using Xamarin.Forms;

namespace GWallet.Frontend.XF.CS
{

    public static class Initialization
    {
        internal static Page LandingPage()
        {
            Page landingPage = new ReceivePage();

            var navPage = new NavigationPage(landingPage);
            NavigationPage.SetHasNavigationBar(landingPage, false);
            return navPage;
        }
    }
}
